pennylane/templates/state_preparations/basis_qutrit.py

```diff
@@ -112,7 +115,23 @@ def compute_decomposition(basis_state, wires):  # pylint: disable=arguments-diff
         """
 
         op_list = []
+
+        if qml.math.is_abstract(basis_state):
+            for wire, state in zip(wires, basis_state):
+                op_list.extend(
+                    [
+                        qml.TRY(state * (2 - state) * np.pi, wires=wire, subspace=(0, 1)),
+                        qml.TRY(state * (1 - state) * np.pi / 2, wires=wire, subspace=(0, 2)),
+                        qml.TRZ((-2 * state + 3) * state * np.pi, wires=wire, subspace=(0, 2)),
+                        qml.TRY(state * (2 - state) * np.pi, wires=wire, subspace=(0, 2)),
+                        qml.TRY(state * (1 - state) * np.pi / 2, wires=wire, subspace=(0, 1)),
+                        qml.TRZ(-(7 * state - 10) * state * np.pi, wires=wire, subspace=(0, 2)),
```

Comment:
Is there a resource for this decomposition?

Reply:
Nope, it came straight from @KetpuntoG's :brain:. We should add extra tests to ensure the JIT decomposition is correct.